route/roundrobin.go

```diff
@@ -48,33 +52,43 @@ func (r *RoundRobin) Next() *Endpoint {
 	return nil
 }
 
-func (r *RoundRobin) next() *endpointElem {
+func (r *RoundRobin) next(attempt int) *endpointElem {
 	r.pool.Lock()
 	defer r.pool.Unlock()
 
+	localDesired := r.locallyOptimistic && attempt == 1
```

Comment:
How is the `attempt` integer indexed? This suggests it's starting from 1, but I've found code that [starts with zero](https://github.com/cloudfoundry/gorouter/pull/382/files?diff=split&w=1#diff-d7673099de14bf52962beecf7bd30f739b20b7a60f70b8a0ef7f4477133e51e6R227-R229).